File: src/main/java/com/kingmang/lazurite/console/output/impl/StringOutput.java
package com.kingmang.lazurite.console.output.impl;

import com.kingmang.lazurite.console.output.Output;
import lombok.AllArgsConstructor;

import java.io.File;

@AllArgsConstructor
public class StringOutput implements Output {

    private final StringBuffer out, err;

    public StringOutput() {
        this(new StringBuffer());
    }

    public StringOutput(StringBuffer out) {
        this(out, out);
    }

    @Override
    public String newline() {
        return System.lineSeparator();
    }

    @Override
    public void print(String value) {
        out.append(value);
    }

    @Override
    public void println() {
        out.append(newline());
    }

    @Override
    public void println(String value) {
        out.append(value).append(newline());
    }

    @Override
    public String toString() {
        return out.toString();
    }

    @Override
    public void error(Throwable throwable) {
        error(throwable.toString());
    }

    @Override
    public void error(CharSequence value) {
        err.append(value).append(newline());
    }

    @Override
    public File fileInstance(String path) {
        return new File(path);
    }
}

File: src/main/java/com/kingmang/lazurite/console/output/impl/StringOutput.kt
package com.kingmang.lazurite.console.output.impl

import com.kingmang.lazurite.console.output.Output
import lombok.AllArgsConstructor
import java.io.File

class StringOutput(private val out: StringBuffer?, private val err: StringBuffer?) : Output {
    constructor(out: StringBuffer? = StringBuffer()) : this(out, out)

    override fun newline(): String =
        System.lineSeparator()

    override fun print(value: String) {
        out!!.append(value)
    }

    override fun println() {
        out!!.append(newline())
    }

    override fun println(value: String) {
        out!!.append(value).append(newline())
    }

    override fun toString(): String {
        return out.toString()
    }

    override fun error(throwable: Throwable) {
        error(throwable.toString())
    }

    override fun error(value: CharSequence) {
        err!!.append(value).append(newline())
    }

    override fun fileInstance(path: String): File =
        File(path)
}